hosts/router/pppoe.nix

```diff
@@ -0,0 +1,40 @@
+{ config , ... }:
+
+{
+  # todo: find a way to load pppoe kernel module
```

Comment:
pppd should do that when it starts

Reply:
I added this because I see this error in the pppd logs
```
Dec 08 14:35:00 router pppd[105605]: Linux kernel does not support PPPoE -- are you running 2.4.x?
```

Reply:
Might actually be a different issue, here is the full log
```
[reimerei@router:~]$ journalctl -fu pppd-dsl.service 
-- Journal begins at Mon 2021-11-15 20:24:28 UTC. --
Dec 08 14:39:59 router pppd[105965]: Warning: can't open options file /root/.ppprc: Permission denied
Dec 08 14:39:59 router pppd[105965]: Plugin rp-pppoe.so loaded.
Dec 08 14:39:59 router pppd[105965]: Plugin rp-pppoe.so loaded.
Dec 08 14:39:59 router pppd[105965]: Couldn't open the /dev/ppp device: No such device or address
Dec 08 14:39:59 router pppd[105965]: Couldn't open the /dev/ppp device: No such device or address
Dec 08 14:39:59 router pppd[105965]: Linux kernel does not support PPPoE -- are you running 2.4.x?
Dec 08 14:39:59 router pppd[105965]: Linux kernel does not support PPPoE -- are you running 2.4.x?
Dec 08 14:39:59 router pppd[105965]: Exit.
Dec 08 14:39:59 router systemd[1]: pppd-dsl.service: Main process exited, code=exited, status=1/FAILURE
Dec 08 14:39:59 router systemd[1]: pppd-dsl.service: Failed with result 'exit-code'.
```

Reply:
It seems to work when I do explicitly load `pppoe`. At least now I'm at the next issue :)